boa/src/builtins/array/mod.rs

```diff
@@ -487,6 +488,45 @@ impl Array {
         Ok(a.into())
     }
 
+    ///'Array.prototype.at(index)'
+    ///
+    /// When at method takes desired integer as index and returns the value at given
+    /// index. Negative integer counts backwards. If index is invalid, the at method
+    /// returns undefined.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-array.prototype.at
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at
+    pub(crate) fn at(this: &JsValue, args: &[JsValue], context: &mut Context) -> JsResult<JsValue> {
+        //1. let O be ? ToObject(this value)
+        let obj = this.to_object(context)?;
+        //2. let len be ? LengthOfArrayLike(O)
+        let len = obj.length_of_array_like(context)? as i64;
+        //3. let relativeIndex be ? ToIntegerOrInfinity(index)
+        let relative_index = args.get(0).unwrap().to_integer_or_infinity(context)?;
+        let k = match relative_index {
+            //4. if relativeIndex >= 0, then let k be relativeIndex
+            IntegerOrInfinity::Integer(i) if i >= 0 => i,
+            //5. Else, let k be len + relativeIndex
+            IntegerOrInfinity::Integer(i) => len + i,
+            //handle most likely impossible case of
+            //IntegerOrInfinity::NegativeInfinity || IntegerOrInfinity::PositiveInfinity
+            //by setting k to len which will return undefined below
+            _ => len,
```

Comment:
Instead of relying on the condition below, maybe it would be better if we early returned here:
```suggestion
            _ => return Ok(JsValue::undefined()),
```
Makes it easier to identify that only finite numbers are valid.

Reply:
Definitely makes more sense to identify only finite numbers being valid